### PR TITLE
[AJ-1416] Fix app count in WDS troubleshooter

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/AJ-1416

While loading the apps list from Leo, the WDS troubleshooter displays "App listing: null app(s) total".

Screenshot 2023-09-19 at 3 17 17 PM

This changes it to show nothing while the number of apps is loading and adds a test to verify that no value is shown for any field while that field is loading.

### CODE REVIEW

File: src/workspace-data/data-table/wds/WdsTroubleshooter.test.ts [hunk 1]
```diff
@@ -174,7 +240,7 @@ describe('WdsTroubleshooter', () => {
     // Assert
     const tableRows = screen.getAllByRole('row');
     const statusLabelAndValueCells = tableRows.map((row) => {
-      const cells = getAllByRole(row, 'cell');
+      const cells = within(row).getAllByRole('cell');
```
Comment (nawatts) [inline src/workspace-data/data-table/wds/WdsTroubleshooter.test.ts:243]: A little refactoring. I think `within(...).getBy...` is clearer.

File: src/workspace-data/data-table/wds/WdsTroubleshooter.ts [hunk 2]
```diff
@@ -83,7 +83,12 @@ export const WdsTroubleshooter = ({ onDismiss, workspaceId, mrgId }) => {
   const troubleShooterText: [string, string | null, boolean, boolean, ReactNode?][] = [
     ['Workspace Id', workspaceId, false, !!workspaceId],
     ['Resource Group Id', mrgId, false, !!mrgId],
-    ['App listing', `${numApps} app(s) total`, numApps == null, !!numApps && numApps !== 'unknown'],
+    [
+      'App listing',
+      numApps === null ? null : `${numApps} app(s) total`,
+      numApps == null,
```
Comment (jladieu) [inline src/workspace-data/data-table/wds/WdsTroubleshooter.ts:89]: Not sure I understand this chain of logic...what's the `numApps == null` clause for?

**Review by jladieu — APPROVED** [review]
One question but LGTM otherwise!

File: src/workspace-data/data-table/wds/WdsTroubleshooter.ts (hunk 2 above)
Comment (nawatts) [reply src/workspace-data/data-table/wds/WdsTroubleshooter.ts:89]: `numApps` is null while the apps list is being loaded from Leo. Currently, this results in the troubleshooter showing "null apps(s) total" during that time. This changes it to render nothing (null) in that row while `numApps` is loading.

Comment (nawatts) [reply src/workspace-data/data-table/wds/WdsTroubleshooter.ts:89]: Oh, I misunderstood the question. That's the logic for determining if the App listing row is still loading.
https://github.com/DataBiosphere/terra-ui/blob/d1438feef5b2739b9d95dc34a8cbe2ac2b04c096/src/workspace-data/data-table/wds/WdsTroubleshooter.ts#L76-L83

**Review by calypsomatic — APPROVED** [review]

**Pull request merged**